editor.planx.uk/src/@planx/components/MultipleFileUpload/Modal.tsx

```diff
@@ -60,3 +72,77 @@ export const FileTaggingModal = (props: FileTaggingModalProps) => {
     </Dialog>
   );
 };
+
+interface SelectMultipleProps extends SelectProps {
+  name: string;
+  fileList: FileList;
+}
+
+const SelectMultiple = (props: SelectMultipleProps) => {
+  const { name, fileList } = props;
+  const [tags, setTags] = React.useState<string[]>([]);
+
+  const handleChange = (event: SelectChangeEvent<typeof tags>) => {
+    const {
+      target: { value },
+    } = event;
+    setTags(
+      // On autofill we get a stringified value.
+      typeof value === "string" ? value.split(",") : value
+    );
+  };
+
+  return (
+    <FormControl key={name} sx={{ display: "flex", flexDirection: "column" }}>
+      <InputLabel id="select-mutliple-file-tags-label">
+        What does this file show?
+      </InputLabel>
+      <Select
+        id="select-multiple-file-tags"
+        labelId="select-multiple-file-tags-label"
+        variant="standard"
+        multiple
+        value={tags}
+        onChange={handleChange}
+        IconComponent={ArrowIcon}
+        input={<Input />}
+        inputProps={{ name }}
+        renderValue={(selected) => selected.join(", ")}
+        MenuProps={{
+          anchorOrigin: {
+            vertical: "bottom",
+            horizontal: "center",
+          },
+        }}
+      >
+        {(Object.keys(fileList) as Array<keyof typeof fileList>).map(
+          (fileListCategory) => (
+            // TODO revisit to conditionally render categories with > 0 files
+            <Box key={`wrapper-${fileListCategory}-files-${name}`}>
+              <ListSubheader
+                key={`subheader-${fileListCategory}-files-${name}`}
+              >
+                {`${capitalize(fileListCategory)} files`}
+              </ListSubheader>
+              {fileList[fileListCategory].map((fileType) => (
+                <MenuItem
+                  key={`menuitem-${fileType.key}-${name}`}
+                  value={fileType.key}
+                >
```

Comment:
While troubleshooting a `handleChange` bug where checklist clicks weren't registering anything, I inspected `MenuItem` and noticed it appears that `key` is only reflecting `menuitem-${name}` which indicates that `fileType.key` is behaving as `undefined` even though it's not ! 

I think it's this error bubbling up again: https://github.com/theopensystemslab/planx-new/pull/1749#discussion_r1219461737

Can someone checkout this branch & double check my hypothesis? If we're seeing this again, shall I rename & update the type throughout to `fileType.name` ?